python/res/enkf/res_config.py

```diff
@@ -66,53 +66,76 @@ class ResConfig(BaseCClass):
     _init_parser       = ResPrototype("void res_config_init_config_parser(config_parser)", bind=False)
 
 
-    def __init__(self, user_config_file=None, config=None, throw_on_error=True):
+    def __init__(self, user_config_file=None, config=None, throw_on_error=True, config_dict=None):
 
+        #for backward compatibility, needs to be supplied in everest as config_dict instead of config
+        if config is not None:
```

Comment:
Why is this needed. Can't we just overtake the usage of config?

Reply:
It's until we make everest ready, since it uses `build_config_content`hence has a bit different structure.

Reply:
Then maybe restructure the init function and split into three sub functions. config_file, config, and config_dict. It may be result in a bit of redundant code, but will make it really a lot more readable

Reply:
ok, have a look whether it is clear enough